GPU Discovery

QuietSwami opened the pull request:
- Added Nvidia Runtime discovery
- Added Dockerfile

QuietSwami committed:
moved development to dev branch

QuietSwami committed:
remove of prints

QuietSwami committed:
cleaning, changes to the dockerfile

QuietSwami committed:
fixing flow

QuietSwami committed:
Merge branch 'master' into dev

QuietSwami committed:
changes to the install of python packages

QuietSwami committed:
Merge branch 'dev' of https://github.com/nuvlabox/peripheral-manager-gpu into dev

QuietSwami committed:
change to the dockerfile

QuietSwami committed:
added version to teh base image

QuietSwami committed:
passing requirement file to the container

Commit:
fix docker cred env var

cjdcordeiro commented on Dockerfile:
```diff
@@ -0,0 +1,8 @@
+FROM ubuntu:18.04
```
please use python:3-alpine

QuietSwami replied:
okay, will change

QuietSwami committed:
changed to alpine

QuietSwami committed:
removed pip install from Dockerfile

cjdcordeiro reviewed (changes requested):
missing compose file

also, please structure the root directory similarly to the other NB components, so that we can clearly identify the code assets and GitHub/Documentation related assets

QuietSwami committed:
created folder structure, docker compose, and updated dockerfile

QuietSwami committed:
created folder structure, docker compose, and updated dockerfile

QuietSwami commented:
> missing compose file
> 
> also, please structure the root directory similarly to the other NB components, so that we can clearly identify the code assets and GitHub/Documentation related assets

Updated with folder structure and docker compose file. Is the version file, in the src folder, needed?

cjdcordeiro commented:
@QuietSwami yes please add it

cjdcordeiro commented on docker-compose.yml:
```diff
@@ -0,0 +1,14 @@
+    build: .
+    restart: on-failure
+    volumes:
+      - /etc/:/etc/
```
missing the :ro

QuietSwami replied:
added

cjdcordeiro commented on docker-compose.yml:
```diff
@@ -0,0 +1,14 @@
+    restart: on-failure
+    volumes:
+      - /etc/:/etc/
+    environment:
```
why do you need these env vars?

QuietSwami replied:
I don't know. I copied the one in modbus, and adapted. is now removed

QuietSwami replied:
We need those env variables to access Nuvla through the API

cjdcordeiro replied:
not anymore. Check the agent API to know how to register the peripherals

QuietSwami commented:
> @QuietSwami yes please add it

Is it the same as the one in modbus?

QuietSwami committed:
added versions.sh and changed docker compose

cjdcordeiro commented:
@QuietSwami yes

QuietSwami committed:
added logging

QuietSwami committed:
version file

QuietSwami committed:
encapsulated the api call in a while tru, trying to fix the bootstrap.

QuietSwami committed:
removed bootstrap

QuietSwami committed:
added log

QuietSwami committed:
fixing logs

QuietSwami committed:
fixing logs

QuietSwami committed:
changed runtime checkup

QuietSwami committed:
removed runtime checkup

QuietSwami committed:
fixed csv line filtre

QuietSwami committed:
checking file output

QuietSwami committed:
added the bootstrap procedure

QuietSwami committed:
debug log for url

QuietSwami committed:
change to the post call

QuietSwami committed:
removed package

QuietSwami committed:
added logging info on the publishing

QuietSwami committed:
testing identifier

QuietSwami committed:
cleaning the dictionary

QuietSwami committed:
checking publishable dictionary

QuietSwami committed:
added availability and classes

QuietSwami committed:
fixing bug while finding docker daemon

QuietSwami committed:
debugging bug in file check

QuietSwami committed:
debugging bug in file check

QuietSwami committed:
check if gpu is already published

QuietSwami committed:
debugging gpu check

QuietSwami committed:
added name to the dictionary

QuietSwami committed:
fixing bug in gpu check

QuietSwami committed:
code cleaning

Commit:
- formatting
 - pep8 fixes
 - removed unused code
 - remove duplicated code
 - clean-ups
 - protected certain variable accesses
 - fix publishing when not needed
 - make script loop indefinitely
 - simplify conditionals
